File: app/src/main/java/com/example/dices2/Cell.java
package com.example.dices2;

import android.content.Context;
import android.view.Gravity;

import java.util.Objects;

public class Cell extends androidx.appcompat.widget.AppCompatTextView {
    private Player owner;
    private RowName row;
    private String value;

    public Cell(Context context, Player owner, RowName row) {
        super(context);
        this.owner = owner;
        this.row = row;
        setTextColor(Consts.MAIN_TABLE_TEXT_COLOR);
        setGravity(Gravity.CENTER_HORIZONTAL);
        setTextSize(Consts.MAIN_TABLE_FONT_SIZE);
    }

    public Player getOwner() {
        return owner;
    }

    public RowName getRow() {
        return row;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
        setText(value);
    }

    public boolean isEmpty() {
        return getText().equals("") || getText() == null;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Cell cell = (Cell) o;
        return owner.equals(cell.owner) && row == cell.row;
    }

    @Override
    public int hashCode() {
        return Objects.hash(owner, row);
    }
}

File: app/src/main/java/com/example/dices2/Cell.kt
package com.example.dices2

import android.content.Context
import android.view.Gravity
import androidx.appcompat.widget.AppCompatTextView
import java.util.*

class Cell(context: Context?, val owner: Player, val row: RowName) : AppCompatTextView(context) {
    var value: String? = null
        set(value) {
            field = value
            text = value
        }

    val isEmpty: Boolean
        get() = text == "" || text == null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val cell = other as Cell
        return owner == cell.owner && row == cell.row
    }

    override fun hashCode(): Int {
        return Objects.hash(owner, row)
    }

    init {
        setTextColor(Consts.MAIN_TABLE_TEXT_COLOR)
        gravity = Gravity.CENTER_HORIZONTAL
        textSize = Consts.MAIN_TABLE_FONT_SIZE.toFloat()
    }
}